src/configure/configurers/provisioningConfigurer.ts

```diff
@@ -0,0 +1,280 @@
+import * as fs from 'fs';
+import * as Path from 'path';
+import * as utils from 'util';
+import * as vscode from 'vscode';
+import { UserCancelledError } from 'vscode-azureextensionui';
+import { ArmRestClient } from '../clients/azure/armRestClient';
+import { IProvisioningServiceClient } from "../clients/IProvisioningServiceClient";
+import { ProvisioningServiceClientFactory } from "../clients/provisioningServiceClientFactory";
+import { sleepForMilliSeconds } from '../helper/commonHelper';
+import { ControlProvider } from '../helper/controlProvider';
+import { GraphHelper } from '../helper/graphHelper';
+import { LocalGitRepoHelper } from '../helper/LocalGitRepoHelper';
+import { telemetryHelper } from '../helper/telemetryHelper';
+import { ExtendedInputDescriptor, InputDataType } from "../model/Contracts";
+import { WizardInputs } from "../model/models";
+import { CompletePipelineConfiguration, DraftPipelineConfiguration, File, ProvisioningConfiguration } from "../model/provisioningConfiguration";
+import { RemotePipelineTemplate } from '../model/templateModels';
+import { Messages } from '../resources/messages';
+import { TelemetryKeys } from '../resources/telemetryKeys';
+import { TracePoints } from '../resources/tracePoints';
+import { InputControl } from '../templateInputHelper/InputControl';
+import { IProvisioningConfigurer } from './IProvisioningConfigurer';
+
+// tslint:disable-next-line:interface-name
+interface DraftFile {
+    content: string;
+    path: string;
+    absPath: string;
+}
+
+const Layer: string = "ProvisioningConfigurer";
+export class ProvisioningConfigurer implements IProvisioningConfigurer {
+    private provisioningServiceClient: IProvisioningServiceClient;
+    private queuedPipelineUrl: string;
+    private refreshTime: number = 5 * 1000;
+    private maxNonStatusRetry: number = 60; // retries for max 5 min
+    private localGitRepoHelper: LocalGitRepoHelper;
+    private filesToCommit: DraftFile[] = [];
+
+    constructor(localGitRepoHelper: LocalGitRepoHelper){
+        this.localGitRepoHelper = localGitRepoHelper;
+    }
+
+    public async queueProvisioningPipelineJob(provisioningConfiguration: ProvisioningConfiguration, wizardInputs: WizardInputs): Promise<ProvisioningConfiguration>{
+        try {
+            this.provisioningServiceClient =  await ProvisioningServiceClientFactory.getClient(wizardInputs.githubPATToken, wizardInputs.azureSession.credentials);
+            const OrgAndRepoDetails = wizardInputs.sourceRepository.repositoryId.split('/');
+            return await this.provisioningServiceClient.createProvisioningConfiguration(provisioningConfiguration, OrgAndRepoDetails[0], OrgAndRepoDetails[1]);
+        } catch (error){
```

Comment:
I am seeing too many try-catch blocks. We should only have try-catch when we want to catch and log telemetry. I don't think we need separate telemetry for all failures. We should have telemetry at a bit higher level. For example, it can be in configurer.

Reply:
Ah this is configurer only. So it is okay.